### PR TITLE
migrate: cast doctests to alloy

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Remove `ethers` from codebase

## Solution

Migrate to `alloy`

### CODE REVIEW

File: crates/cast/src/lib.rs [hunk 1]
```diff
@@ -88,25 +88,27 @@ where
...
+    /// ```
+    /// use alloy_primitives::{Address, U256, Bytes};
+    /// use alloy_rpc_types::{TransactionRequest, WithOtherFields};
+    /// use cast::Cast;
     /// use foundry_common::provider::alloy::get_http_provider;
     /// use std::str::FromStr;
+    /// use alloy_sol_types::{sol, SolCall};
+    ///
+    /// sol!(
+    /// function greeting(uint256 i) public returns (string);
```
Comment (onbjerg) [inline crates/cast/src/lib.rs:100]:
```suggestion
    ///     function greeting(uint256 i) public returns (string);
```
nit

File: crates/cast/src/lib.rs [hunk 2]
```diff
@@ -159,23 +161,27 @@ where
...
+    /// ```
+    /// use cast::{Cast};
+    /// use alloy_primitives::{Address, U256, Bytes};
+    /// use alloy_rpc_types::{TransactionRequest, WithOtherFields};
+    /// use foundry_common::provider::alloy::get_http_provider;
     /// use std::str::FromStr;
+    /// use alloy_sol_types::{sol, SolCall};
+    ///
+    /// sol!(
+    /// function greeting(uint256 i) public returns (string);
```
Comment (onbjerg) [inline crates/cast/src/lib.rs:173]:
```suggestion
    ///     function greeting(uint256 i) public returns (string);
```
nit

File: crates/cast/src/lib.rs [hunk 3]
```diff
@@ -215,27 +221,32 @@ where
...
+    /// ```
+    /// use cast::{Cast};
+    /// use alloy_primitives::{Address, U256, Bytes};
+    /// use alloy_rpc_types::{TransactionRequest, WithOtherFields};
+    /// use foundry_common::provider::alloy::get_http_provider;
     /// use std::str::FromStr;
+    /// use alloy_sol_types::{sol, SolCall};
+    ///
+    /// sol!(
+    /// function greet(string greeting) public;
```
Comment (onbjerg) [inline crates/cast/src/lib.rs:233]:
```suggestion
    ///     function greeting(uint256 i) public returns (string);
```
nit

**Review by onbjerg — APPROVED** [review]
smol nits lgtm otherwise

File: crates/cast/src/lib.rs [hunk 4]
```diff
@@ -159,23 +161,27 @@ where
...
     /// use std::str::FromStr;
+    /// use alloy_sol_types::{sol, SolCall};
+    ///
+    /// sol!(
+    /// function greeting(uint256 i) public returns (string);
+    /// );
     ///
     /// # async fn foo() -> eyre::Result<()> {
-    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
+    /// let provider = get_http_provider("http://localhost:8545");
```
Comment (mattsse) [inline crates/cast/src/lib.rs:177]: this should not be part of the docs, this is an internal helper function

the example should just use a regular alloy provider

File: crates/cast/src/lib.rs [hunk 5]
```diff
@@ -215,27 +221,32 @@ where
...
+    /// function greet(string greeting) public;
+    /// );
     ///
     /// # async fn foo() -> eyre::Result<()> {
-    /// let provider = Provider::<Http>::try_from("http://localhost:8545")?;
-    /// let from = "vitalik.eth";
-    /// let to = eAddress::from_str("0xB3C95ff08316fb2F2e3E52Ee82F8e7b605Aa1304")?;
-    /// let sig = "greet(string)()";
-    /// let args = vec!["hello".to_owned()];
+    /// let provider = get_http_provider("http://localhost:8545");
```
Comment (mattsse) [inline crates/cast/src/lib.rs:237]: same

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by DaniPopes — APPROVED** [review]

**Review by ArkFunds — APPROVED** [review]
True

**Review by ArkFunds — APPROVED** [review]
TRUE

**Pull request merged**